### PR TITLE
paste documentation

### PR DESCRIPTION
**Pull request by shannonpileggi**

### CODE REVIEW

**Comment by shannonpileggi** [comment]
There is healthy discussion of `paste` documentation in [bugzilla](https://bugs.r-project.org/show_bug.cgi?id=17933). In part because of some the discussion mixes effects of same length vs different length vectors (i.e., in the presence of recycling vs not). In addition, the sole zero-length input has special behavior.

Here are the cases I explored:

```
# no recycling; single input
paste({}, recycle0 = TRUE, collapse = NULL, sep = "|")
paste({}, recycle0 = TRUE, collapse = "&", sep = "|")
paste({}, recycle0 = FALSE, collapse = NULL, sep = "|")
paste({}, recycle0 = FALSE, collapse = "&", sep = "|")

# no recycling; two inputs
paste({}, 1, recycle0 = TRUE, collapse = NULL, sep = "|")
paste({}, 1, recycle0 = TRUE, collapse = "&", sep = "|")
paste({}, 1, recycle0 = FALSE, collapse = NULL, sep = "|")
paste({}, 1, recycle0 = FALSE, collapse = "&", sep = "|")

# recycling; two inputs
paste({}, 1:2, recycle0 = TRUE, collapse = NULL, sep = "|")
paste({}, 1:2, recycle0 = TRUE, collapse = "&", sep = "|")
paste({}, 1:2, recycle0 = FALSE, collapse = NULL, sep = "|")
paste({}, 1:2, recycle0 = FALSE, collapse = "&", sep = "|")
```

with results summarized as follows:
image

In order to clarify documentation, I propose 
1. Modifying the "Details" section
2. Modifying the arguments description of `recycle0` 
3. Modifying the examples section

File: src/library/base/man/paste.Rd [hunk 1]
```diff
@@ -19,23 +19,24 @@ paste0(\dots,            collapse = NULL, recycle0 = FALSE)
...
   \item{recycle0}{\code{\link{logical}} indicating if zero-length
-    character arguments should lead to the zero-length
-    \code{\link{character}(0)} after the \code{sep}-phase (which turns into
-    \code{""} in the \code{collapse}-phase, i.e., when \code{collapse} is
-    not \code{NULL}).}
+    character arguments should result in the zero-length
+    \code{\link{character}(0)}. This argument interacts with \code{collapse}
+    such that under the conditions \code{recycle0} is \code{TRUE} and
+    \code{collapse} is not \code{NULL}, zero-length arguments instead output as
+    \code{""} .}
```
Comment (trangdata) [inline src/library/base/man/paste.Rd]: A minor suggestion on spacing
```suggestion
    \code{""}.}
```

**Review by trangdata — APPROVED** [review]
Looks good to me! 💯  I only have a minor comment.

File: src/library/base/man/paste.Rd (hunk 1 above)
Comment (shannonpileggi) [reply src/library/base/man/paste.Rd]: excellent catch, thank you so much @trangdata!

File: src/library/base/man/paste.Rd [hunk 2]
```diff
@@ -120,18 +121,19 @@ paste(month.abb, nth, sep = ": ", collapse = "; ")
     width = 30), collapse = "\n"))
 plot(dist ~ speed, cars, main = title)
 
-## 'recycle0 = TRUE' allows more vectorized behaviour, i.e. zero-length recycling :
-valid <- FALSE
-val <- pi
-paste("The value is", val[valid], "-- not so good!")
-paste("The value is", val[valid], "-- good: empty!", recycle0=TRUE) # -> character(0)
+# zero length arguments recycled as `""`
+paste({}, 1:2)
```
Comment (MichaelChirico) [inline src/library/base/man/paste.Rd]: Personally I would prefer `paste(character(), 1:2)`. `{}` is concise, but masks two behaviors:

 1. [the empty expression evaluates to `NULL`](https://github.com/r-devel/r-svn/blob/3c24c211df69cf01c61ec61e5de5ad053cdf1f1c/src/main/eval.c#L2917-L2939): `identical(NULL, {})` is `TRUE`
 2. [inputs to `paste()` are cast with `as.character()`](https://github.com/r-devel/r-svn/blob/3c24c211df69cf01c61ec61e5de5ad053cdf1f1c/src/main/paste.c#L147-L161): `identical(as.character(NULL), character())` is `TRUE`
 
 Using `character()` is thus more "minimal" in the sense of demonstrating the behavior more directly

Comment (shannonpileggi) [reply src/library/base/man/paste.Rd]: Thank you for pointing this out, @MichaelChirico!

**Pull request closed without merging**